CMakeLists.txt

```diff
@@ -12,6 +12,21 @@ if (NOT CMAKE_BUILD_TYPE)
             "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
 endif (NOT CMAKE_BUILD_TYPE)
 
+# set optimization flags for release build
+if (CMAKE_BUILD_TYPE STREQUAL "Release")
+    set(CMAKE_CXX_FLAGS_RELEASE "-O2" CACHE STRING "Release build optimization level" FORCE)
+endif ()
+
+# Enable AdressSanitizer for Debug builds (Continuous Integration)
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address -fsanitize=undefined")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address")
+endif ()
```

Comment:
Be aware: the address sanitizer slows down the code. You may want to allow for a build type which can be used for debugging without the address sanitizer.